File: app/src/main/java/com/example/sampledax/ui/dax/data/response/TopList.java

package com.example.sampledax.ui.dax.data.response;


import com.google.gson.annotations.SerializedName;


@SuppressWarnings("unused")
public class TopList {

    @SerializedName("cnDivYield")
    private Double mCnDivYield;
    @SerializedName("cnPeg")
    private Double mCnPeg;
    @SerializedName("cnPer")
    private Double mCnPer;
    @SerializedName("datetimeLast")
    private String mDatetimeLast;
    @SerializedName("idNotation")
    private Long mIdNotation;
    @SerializedName("instrument")
    private Instrument mInstrument;
    @SerializedName("isoCurrency")
    private String mIsoCurrency;
    @SerializedName("last")
    private Double mLast;
    @SerializedName("money")
    private Double mMoney;
    @SerializedName("performance")
    private Double mPerformance;
    @SerializedName("performancePct")
    private Double mPerformancePct;
    @SerializedName("volume")
    private Double mVolume;

    public Double getCnDivYield() {
        return mCnDivYield;
    }

    public void setCnDivYield(Double cnDivYield) {
        mCnDivYield = cnDivYield;
    }

    public Double getCnPeg() {
        return mCnPeg;
    }

    public void setCnPeg(Double cnPeg) {
        mCnPeg = cnPeg;
    }

    public Double getCnPer() {
        return mCnPer;
    }

    public void setCnPer(Double cnPer) {
        mCnPer = cnPer;
    }

    public String getDatetimeLast() {
        return mDatetimeLast;
    }

    public void setDatetimeLast(String datetimeLast) {
        mDatetimeLast = datetimeLast;
    }

    public Long getIdNotation() {
        return mIdNotation;
    }

    public void setIdNotation(Long idNotation) {
        mIdNotation = idNotation;
    }

    public Instrument getInstrument() {
        return mInstrument;
    }

    public void setInstrument(Instrument instrument) {
        mInstrument = instrument;
    }

    public String getIsoCurrency() {
        return mIsoCurrency;
    }

    public void setIsoCurrency(String isoCurrency) {
        mIsoCurrency = isoCurrency;
    }

    public Double getLast() {
        return mLast;
    }

    public void setLast(Double last) {
        mLast = last;
    }

    public Double getMoney() {
        return mMoney;
    }

    public void setMoney(Double money) {
        mMoney = money;
    }

    public Double getPerformance() {
        return mPerformance;
    }

    public void setPerformance(Double performance) {
        mPerformance = performance;
    }

    public Double getPerformancePct() {
        return mPerformancePct;
    }

    public void setPerformancePct(Double performancePct) {
        mPerformancePct = performancePct;
    }

    public Double getVolume() {
        return mVolume;
    }

    public void setVolume(Double volume) {
        mVolume = volume;
    }

}

File: app/src/main/java/com/example/sampledax/ui/dax/data/response/TopList.kt
package com.example.sampledax.ui.dax.data.response

import com.google.gson.annotations.SerializedName
import kotlin.collections.List


@Suppress("unused")
class TopList {
    @SerializedName("cnDivYield")
    var cnDivYield: Double? = null

    @SerializedName("cnPeg")
    var cnPeg: Double? = null

    @SerializedName("cnPer")
    var cnPer: Double? = null

    @SerializedName("datetimeLast")
    var datetimeLast: String? = null

    @SerializedName("idNotation")
    var idNotation: Long? = null

    @SerializedName("instrument")
    var instrument: Instrument? = null

    @SerializedName("isoCurrency")
    var isoCurrency: String? = null

    @SerializedName("last")
    var last: Double? = null

    @SerializedName("money")
    var money: Double? = null

    @SerializedName("performance")
    var performance: Double? = null

    @SerializedName("performancePct")
    var performancePct: Double? = null

    @SerializedName("volume")
    var volume: Double? = null

    class Instrument {
        @SerializedName("displayType")
        var displayType: String? = null

        @SerializedName("entityAttributes")
        var entityAttributes: List<String>? = null

        @SerializedName("entityType")
        var entityType: String? = null

        @SerializedName("entityValue")
        var entityValue: String? = null

        @SerializedName("homeSymbol")
        var homeSymbol: String? = null

        @SerializedName("instrumentType")
        var instrumentType: String? = null

        @SerializedName("isin")
        var isin: String? = null

        @SerializedName("name")
        var name: String? = null

        @SerializedName("symbol")
        var symbol: String? = null

        @SerializedName("tinyName")
        var tinyName: String? = null

        @SerializedName("type")
        var type: String? = null

        @SerializedName("underlyingEntity")
        var underlyingEntity: UnderlyingEntity? = null

        @SerializedName("urlName")
        var urlName: String? = null

        @SerializedName("urls")
        var urls: Urls? = null

        @SerializedName("wkn")
        var wkn: String? = null
    }
}
